### PR TITLE
feat: Network flood control

### PR DESCRIPTION
**Pull request by TheWaWaR**
Do not send blocks to peer when session send buffer is full.

### CODE REVIEW

File: network/src/config.rs [hunk 1]
```diff
@@ -36,6 +38,9 @@ pub struct NetworkConfig {
     pub upnp: bool,
     #[serde(default)]
     pub bootnode_mode: bool,
+    // Max send buffer size
+    #[serde(default)]
```
Comment (doitian) [inline network/src/config.rs]: Not necessary, since Option is allowed omit in the toml.

File: network/src/protocols/mod.rs [hunk 2]
```diff
@@ -58,6 +58,7 @@ pub trait CKBProtocolContext: Send {
     fn connected_peers(&self) -> Vec<PeerIndex>;
     fn report_peer(&self, peer_index: PeerIndex, behaviour: Behaviour);
     fn ban_peer(&self, peer_index: PeerIndex, duration: Duration);
+    fn pause_send(&self) -> bool;
```
Comment (doitian) [inline network/src/protocols/mod.rs]: It looks like a method which performs an action to pause sending. `send_paused` is a better name.

Comment (TheWaWaR) [reply network/src/protocols/mod.rs]: `pause_send` is from bitcoin, but `send_paused` seems more readable.

**Comment by TheWaWaR** [comment]
@doitian 
Can we merge this now?

**Pull request merged**